File: Java/Technology Fundamentals with Java/Exercises Java Code/src/MethodExercise/MiddleCharacters.java
package MethodExercise;

import java.util.Scanner;

public class MiddleCharacters {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String text = sc.nextLine();

        printMiddleCharacter(text);

        System.out.println();
    }

    private static void printMiddleCharacter(String text) {
        if (text.length()%2==0){
            System.out.println(text.charAt(text.length()/2-1)+""+text.charAt(text.length()/2));
        }
        else{
            System.out.println(text.charAt(text.length()/2));
        }
    }
}

File: C#/Technology Fundamentals with C#/04. Methods - Functions - Exercise/06. Middle Characters/MiddleCharacters.cs
using System;

public class MiddleCharacters
{
    public static void Main()
    {
        string text = Console.ReadLine();

        PrintMiddleChar(text);
    }

    private static void PrintMiddleChar(string text)
    {
        if (text.Length % 2 == 0)
        {
            Console.WriteLine(text[text.Length / 2 - 1] + "" + text[text.Length / 2]);
        }
        else
        {
            Console.WriteLine(text[text.Length / 2]);
        }
    }
}
